File: boot/debug.js
const fs = require('fs');
const elfy = require('elfy');

// Path to the kernel ELF file
const elfFilePath = 'build/kernel.o';

// Read the ELF file
fs.readFile(elfFilePath, (err, data) => {
    if (err) {
        console.error('Error reading ELF file:', err);
        return;
    }

    console.log(data);

    // Parse the ELF file
    const parsedElf = elfy.parse(data);

    // Find the section containing the kernel signature
    const signatureSection = parsedElf.sections.find(section => section.name === '.kernel_signature');
    if (signatureSection) {
        // Extract the kernel signature data
        const signatureData = data.slice(signatureSection.offset, signatureSection.offset + signatureSection.size);

        // Convert the signature data to a hexadecimal string
        const signatureHex = signatureData.toString('hex');

        // Print the kernel signature
        console.log('Kernel Signature:', signatureHex);
    } else {
        console.log('Kernel signature section not found in the ELF file.');
    }
});

File: boot/debug.py
import elftools.elf.elffile as elf

# Path to the kernel ELF file
elf_file_path = 'build/kernel.o'

# Open the ELF file
with open(elf_file_path, 'rb') as f:
    # Parse the ELF file
    elf_file = elf.ELFFile(f)
    
    # Get the section containing the kernel signature
    signature_section = elf_file.get_section_by_name('.kernel_signature')
    
    if signature_section:
        # Read the kernel signature data
        signature_data = signature_section.data()
        
        # Convert the signature data to a hexadecimal string
        signature_hex = ''.join(format(byte, '02x') for byte in signature_data)
        
        # Print the kernel signature
        print("Kernel Signature:", signature_hex)
    else:
        print("Kernel signature section not found in the ELF file.")
